magento/magento2
# Performance - Optimize getSpecificOptions triggered by all getAttributeText calls

### Description (*)
When performing a call to `$product->getAttributeText('my_attribute')` to output the text value of the attribute, it always trigger a call to database. I introduce a memoization to save the previously calculated result and avoid to call database each time for the same combination $storeId / $attributeId / $optionId

### Related Pull Requests


### Fixed Issues (if relevant)
1. Fixes magento/magento2#36174

### Manual testing scenarios (*)
1. Go to a page where getAttributeText is used depending of your templates files (category page, product page), for example, use `getAttributeText('color')`
2. If you have two or more products having the exact same color, only one call will be made to database
3. You can notice it by using Blackfire or a log in the newly introduced method `loadSpecificOptions`

### Questions or comments

### Contribution checklist (*)
 - [*] Pull request has a meaningful description of its purpose
